ui/src/wallet.ts

```diff
@@ -0,0 +1,135 @@
+import WalletConnect from "@walletconnect/client";
+import { convertUtf8ToHex } from "@walletconnect/utils";
+import QRCodeModal from "@walletconnect/qrcode-modal";
+import WalletConnectWeb3Provider from "@walletconnect/web3-provider";
+import Web3 from "web3";
+import { writable, Readable } from "svelte/store";
+
+export enum Status {
+  Connected = "CONNECTED",
+  NotConnected = "NOT_CONNECTED",
+}
+
+type State =
+  | { status: Status.NotConnected }
+  | { status: Status.Connected; connected: Connected };
+
+interface Connected {
+  chainId: number;
+  account: {
+    address: string | null;
+    balance: string | null;
+  };
+}
+
+interface Wallet extends Readable<State> {
+  connect(): Promise<void>;
+  disconnect(): Promise<void>;
+  testSign(): Promise<void>;
+}
+
+export function makeWallet(): Wallet {
```

Comment:
Is this constructor adding much value? It's called once in the wallet component and otherwise not used in other ways. I think removing it and just export the relevant parts from the module directly is as ergonomic and the more widely used style in the rest of the code-base.

Reply:
If we don't have it as a "constructor" we can't build a new instance as needed, which now is necessary since we face issues when disconnecting, were creating a new instance is a useful workaround for now.

Reply:
Btw, in the future, we want to offer more alternatives to WalletConnect to I will move this to a `walletconnect.ts` module and have this constructor receive a "ConnectorOption` or so and call the appropriate constructor.